mythril/laser/ethereum/instructions.py

```diff
@@ -1012,15 +1046,22 @@ def jump_(self, global_state: GlobalState) -> List[GlobalState]:
             )
 
         new_state = copy(global_state)
+        # add JUMP gas cost
+        min_gas, max_gas = OPCODE_GAS["JUMP"]
```

Comment:
Shouldn't you still pay gas, even if the jump is invalid?

Reply:
Yeah, true. On an invalid jump, the whole gas is actually burnt to punish the user. From the yellowpaper:

> Just as with contract creation, if the execution halts in an exceptional fashion (i.e.  due to an exhausted gas supply, stack underflow, invalid jump destination or invalid instruction), then no gas is refunded to the caller and the state is reverted to the point immediately prior to balance transfer (i.e. σ).

Reply:
So this should in theory be above the jumpdest check? (Not that it really matters for laser)

Reply:
It would burn the gas after jumpdest check in case it failed :)